proto/process/connections.proto

```diff
@@ -184,6 +184,9 @@ message Connection {
 
   // serialized HTTP2Aggregations object summarizing all http2 transactions recorded for this connection, organized by request path
   bytes http2Aggregations = 50;
+
+  // serialized map of POSIX error codes (ex ETIMEOUT) to the number of times they have been encountered in the context of this connection
```

Comment:
are these codes meant to be platform dependent, or is windows meant to approximate the POSIX error codes ?

Reply:
How would you all feel about using a our own logical name for the error codes instead of potentially different Windows / Linux codes that mean the same thing? Instead of ETIMEOUT maybe represent it to the customer as DDTIMEOUT or something, just in case there is a difference between Windows / Linux. More of an idea than a suggestion - curious to hear thoughts.

Reply:
If we come up with our own error code scheme, I'd recommend using an int enum instead of string as it will serialize more efficiently

Reply:
Actually come to think of it I don't like the idea of the map key here being a string. I think this should be an int or an enum. enum seems more appealing to me, which means we should define our own codes. On linux they can map to POSIX error codes; Windows can do the same with whatever scheme they want to use. This would also make it easier for the backend to interpret these errors.

Reply:
works for me, I'll make the change

Reply:
This would be the reason to use our enums, which is what I was trying to say in my last comment.

> The thing that is important to me here is that a failed TCP client connection should be displayed to the customer in the same way on Windows or on Linux. For example, if the connection is refused it should be CONNECTION_REFUSED and if the connection times out, it should be CONNECTION_TIMEDOUT, regardless of it came from Windows / Linux and regardless of how it was collected on Windows / Linux.

Reply:
To address a few points:
 - I think the payload data should be platform agnostic. Windows and linux should set the same map keys and values, keeping an internal mapping if necessary.
 - In protobuf, enums can't be used as map keys. So we're left with using ints or strings.

 - I think it makes more sense to use ints. In linux, the errnos are ints (ie `ETIMEOUT` is `110`). Storing them as strings in the payload means we have to map them from int -> string in the agent in userspace and store them that way on heap. Errnos are <256 I think, so we could even use int8 to store them in the agent and payload (which uses varint). At the scale of every connection, storing ints saves a decent amount of space vs strings.
 - If we use ints, windows would have to map whatever the internal representation of an error is from that to the corresponding errno int. I think this is cheap

What does everyone thing? cc: @leeavital @hmahmood @richl-dd